modules/image_dataset.ipynb

```diff
@@ -0,0 +1,234 @@
+    "# limitations under the License.\n",
+    "import os\n",
+    "import tempfile\n",
+    "import shutil\n",
+    "\n",
+    "import nibabel as nib\n",
+    "import numpy as np\n",
+    "\n",
+    "from monai.data import ImageDataset\n",
+    "from monai.transforms import Compose, EnsureChannelFirst, RandAdjustContrast, Spacing\n",
+    "from monai.config import print_config\n",
+    "\n",
+    "print_config()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Generate some image data\n",
+    "tempdir = tempfile.mkdtemp()\n",
+    "img_ = nib.Nifti1Image(np.random.randint(0, 2, size=(20, 20, 20)), np.eye(4))\n",
+    "seg_ = nib.Nifti1Image(np.random.randint(0, 2, size=(20, 20, 20)), np.eye(4))\n",
+    "img_name = os.path.join(tempdir, \"img.nii.gz\")\n",
+    "seg_name = os.path.join(tempdir, \"seg.nii.gz\")\n",
+    "nib.save(img_, img_name)\n",
+    "nib.save(seg_, seg_name)\n",
+    "img_list = [img_name]\n",
+    "seg_list = [seg_name]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Build a customised chain of transforms\n",
+    "`TestCompose` is used to handle the additional metadata as a part of the preprocessing pipeline."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2021-01-14T12:34:07.530773Z",
+     "iopub.status.busy": "2021-01-14T12:34:07.530335Z",
+     "iopub.status.idle": "2021-01-14T12:34:08.080092Z",
+     "shell.execute_reply": "2021-01-14T12:34:08.080424Z"
+    },
+    "tags": []
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "image shape: (1, 14, 14, 7)\n",
+      "seg. shape: (1, 14, 14, 7)\n"
+     ]
+    }
+   ],
+   "source": [
+    "class TestCompose(Compose):\n",
+    "    def __call__(self, input_):\n",
+    "        data, meta = input_\n",
+    "        data = self.transforms[0](data, meta)  # ensure channel first\n",
+    "        data, _, meta[\"affine\"] = self.transforms[1](data, meta[\"affine\"])  # spacing\n",
+    "        if len(self.transforms) == 3:\n",
+    "            return self.transforms[2](data), meta  # image contrast\n",
+    "        return data, meta\n",
+    "\n",
+    "\n",
+    "img_xform = TestCompose([EnsureChannelFirst(), Spacing(pixdim=(1.5, 1.5, 3.0)), RandAdjustContrast()])\n",
+    "seg_xform = TestCompose([EnsureChannelFirst(), Spacing(pixdim=(1.5, 1.5, 3.0), mode=\"nearest\")])\n",
+    "img_dataset = ImageDataset(\n",
+    "    image_files=img_list,\n",
+    "    seg_files=seg_list,\n",
+    "    transform=img_xform,\n",
+    "    seg_transform=seg_xform,\n",
+    "    image_only=False,\n",
+    "    transform_with_metadata=True,\n",
+    ")\n",
+    "print(\"image shape:\", img_dataset[0][0].shape)\n",
+    "print(\"seg. shape:\", img_dataset[0][1].shape)"
```

Comment:
Please remove this typo "."

Reply:
the 'seg.' is an abbreviation of segmentation